### PR TITLE
Update core package to support v2 API contract

### PR DESCRIPTION
**Pull request by shenglol**
- Added v2 API models and interface
- Added validators that extensibility providers can reuse to simplify model validation

### CODE REVIEW

File: ' [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (majastrz) [inline ':1]: Is this file name intentional?

File: src/Azure.Deployments.Extensibility.Core.Tests.Unit/Azure - Backup.Deployments.Extensibility.Core.Tests.Unit.csproj [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (majastrz) [inline src/Azure.Deployments.Extensibility.Core.Tests.Unit/Azure - Backup.Deployments.Extensibility.Core.Tests.Unit.csproj:1]: >Project [](http://example.com/codeflow?start=1&length=7)

Spaces in the file name intentional?

**Review by majastrz — APPROVED** [review]
:shipit:

File: ' (hunk 1 above)
Comment (shenglol) [reply ']: No, I don't know how this got created. I don't know if it's a bug - I've encountered instances where similar files were generated randomly while using Visual Studio.

Comment (shenglol) [reply ']: Removed it.

File: src/Azure.Deployments.Extensibility.Core.Tests.Unit/Azure - Backup.Deployments.Extensibility.Core.Tests.Unit.csproj (hunk 2 above)
Comment (shenglol) [reply src/Azure.Deployments.Extensibility.Core.Tests.Unit/Azure - Backup.Deployments.Extensibility.Core.Tests.Unit.csproj:1]: Hmm I wonder why VS inserted the spaces. I created the project in VS without checking the file system. Let me fix this.

Comment (shenglol) [reply src/Azure.Deployments.Extensibility.Core.Tests.Unit/Azure - Backup.Deployments.Extensibility.Core.Tests.Unit.csproj:1]: Wait this is a back up file. Will remove it.

**Pull request merged**